### PR TITLE
Upload exception message to Sentry

### PR DESCRIPTION
**Pull request by mallexxx**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1202406491309510/1207597188072658/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2974
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2886
What kind of version bump will this require?: Major

**Optional**:

Tech Design URL: https://app.asana.com/0/1202406491309510/1207311510824180/f

**Description**:
- unhandled NSException handler
- unhandled C++ exception saving call stack and message
- writing sanitized diagnostics to log file
- uploading diagnostics to Sentry

**Steps to test this PR**:
1. Build/Install the app;
2. Launch app (standalone), open Debug menu, simulate NSException crash/C++ exception crash
3. Launch app from Xcode, validate diagnostic message is loaded and uploaded to Sentry

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/Common/Extensions/StringExtension.swift [hunk 1]
```diff
@@ -93,6 +95,235 @@ public extension String {
...
+    }
+
+    private func findFilePathEnd(from pathStartIdx: String.Index) -> String.Index {
+        // macOS file names can contain literally any Unicode character except `/` and newline with max length=255
+        // but let‘s assume some general naming conventions:
+        // - `filename.extension` followed by a word boundary [ ,.:;], not `/`, terminates the file path
+        // - file/folder names should not contain trailing spaces
+        //   although technically it‘s possible, but if the next path component starts with `/` we'll treat it as another path
+        //
+        // 1. find end of the line
```
Comment (bwaresiak) [inline Sources/Common/Extensions/StringExtension.swift:300]: I winder if we are a bit generous here, e.g. `"Error in /var/filename.txt In file included from /Volumes/data/framework dir".sanitized()` returns `Error in <removed>` 

I'd love to see more tests for simple strings that showcase what is expected to match and what not, with a bit clearer test names :)

File: Package.swift [hunk 2]
```diff
@@ -21,6 +21,7 @@ let package = Package(
         .library(name: "BloomFilterWrapper", targets: ["BloomFilterWrapper"]),
         .library(name: "UserScript", targets: ["UserScript"]),
         .library(name: "Crashes", targets: ["Crashes"]),
+        .library(name: "cxxCrashHandler", targets: ["cxxCrashHandler"]),
```
Comment (bwaresiak) [inline Package.swift:24]: Xcode complaints about naming on my machine:

cxx

Comment (mallexxx) [reply Package.swift]: I double renamed it to a capitalized form, could you re-check

**Review by bwaresiak — APPROVED** [review]
Great job @mallexxx :)

**Pull request merged**